### ISSUE TITLE
Contracting snippets at the end of a file causes tabout and bugs the editor

### ISSUE THREAD

**Issue by vlad2305m** [artisticat1/obsidian-latex-suite#70]
Snippets like {"^>"→"⟳"}, aka contracting snippets cause Tab to act on the browser interface (and tab outside the editor) instead of eating it, which makes the editor pane go crazy (into some weird insert-mode, tuning off saving the document once bugged).
Currently investigating further...

**Comment by vlad2305m** [comment]
Maybe it happens when cursor ends up outside the file, but `app.vault.modify()` works fine with that when I use Templater. 
```
RangeError: Invalid change range 94 to 96 (in doc of length 95)
    at e (app.js:1:272982)
    at e (app.js:1:272725)
    at Function.t.of (app.js:1:273231)
    at SnippetManager.expandSnippets (plugin:obsidian-latex-suite:7783:47)
    at LatexSuitePlugin.runSnippets (plugin:obsidian-latex-suite:8177:43)
    at LatexSuitePlugin.handleKeydown (plugin:obsidian-latex-suite:8101:26)
    at Object.LatexSuitePlugin.onKeydown (plugin:obsidian-latex-suite:8086:28)
    at e.runCustomHandlers (app.js:1:376149)
    at HTMLDivElement.<anonymous> (app.js:1:374890)
ro @ app.js:1
e.runCustomHandlers @ app.js:1
(anonymous) @ app.js:1
```

**Comment by artisticat1** [comment]
Hi! Thanks for the report. I don't notice any issues with the snippet

```typescript
{trigger: "^>", replacement: "⟳", options: "A"}
```

However, after removing the "auto" option so that the snippet has to be triggered with Tab, I find that pressing Tab to trigger the snippet does cause you to tab out of the editor, which is very strange. What do you mean by "contracting snippets"?

I'll look into the issue and your proposed solution when I have time!

**Comment by vlad2305m** [comment]
I never use automatic snippets )
Feel like Tab is better than Ctrl+z

**Comment by artisticat1** [comment]
Fixed in 1.6.2, thanks for providing the solution!
